app/jobs/cleanup_account_job.rb

```diff
@@ -1,18 +1,32 @@
 class CleanupAccountJob < ActiveJob::Base
   def perform(account)
-    cleanup_solr(account)
-    cleanup_fedora(account)
+    # Ensure all account endpoints are active/configured
+    account.switch do
+      cleanup_solr(account)
+      cleanup_fedora(account)
+      # Redis doesn't need 'account', as switch changes Redis::Namespace to correct namespace
+      cleanup_redis
+    end
     Apartment::Tenant.drop(account.tenant)
     account.destroy
   end
 
   private
 
     def cleanup_fedora(account)
-      account.switch do
-        # Preceding slash must be removed from base_path when calling delete()
-        fcrepo_client.delete(account.fcrepo_endpoint.base_path.sub!(%r{^\/}, ''))
-      end
+      # Preceding slash must be removed from base_path when calling delete()
+      fcrepo_client.delete(account.fcrepo_endpoint.base_path.sub!(%r{^\/}, ''))
```

Comment:
You shouldn't need the backslash here. That's part of why the `%r{ }` construct is useful!